docs/walkthrough.md

```diff
@@ -0,0 +1,500 @@
+Walkthrough
+===========
+
+This walkthrough will go over the basics of setting up the Prometheus
+adapter on your cluster and configuring an autoscaler to use application
+metrics sourced from the adapter.
+
+Prerequisites
+-------------
+
+### Cluster Configuration ###
+
+Before getting started, ensure that the main components of your
+cluster are configured for autoscaling on custom metrics.  As of
+Kubernetes 1.7, this requires enabling the aggregation layer on the API
```

Comment:
Tell about the args that are required on the API server and the certs that have to be provided there in order for this to work. You can link to the API Aggregation docs as well, and tell users that kubeadm does this OOTB :)